shell/server/drivers/mail.js

```diff
@@ -223,15 +240,23 @@ hackSendEmail = (session, email) => {
       };
     }
 
-    if (email.from.address !== grainAddress && email.from.address !== userAddress.address) {
+    if (removeValidPlusSuffix(email.from.address) !== grainAddress && email.from.address !== userAddress.address) {
       throw new Error(
-        "FROM header in outgoing emails need to equal either " + grainAddress + " or " +
+        "FROM header in outgoing emails need to equal either " + grainAddress + " (with optional suffix) or " +
         userAddress.address + ". Yours was: " + email.from.address);
     }
 
     // Unpack fields
     const { from, to, cc, bcc, replyTo, subject, text, html } = email;
 
+    // We allow envelope address to have a suffix as well. In this way apps can use unique envelope
+    // addresses when sending out e-mails which allow better handling of e-mail bounces - app can
+    // match bounce with the sent e-mail.
+    let envelopeFrom = email.envelopeFrom || from.address;
```

Comment:
OK, I agree about long term goals here. But my question what can I do in short term, until all that is in place? It seems for now the best way to have github-like notification is to implement them myself (I will have them anyway for running app outside Sandstorm) and get small changes in Sandstorm in place to allow this. But then it seems it will not work anyway on Oasis because of Sendgrid and general sending limits and lack of background e-mail sending context or in general background tasks. So maybe I should just give up with this approach and wait for Sandstorm to have notifications API and until then the app will not have this feature when running inside Sandstorm.